src/services/report_service.js

```diff
@@ -43,12 +44,23 @@ export default (easypostClient) =>
      * Retrieve the next page of specific collection of object
      * @param {Object} reports An object containing a list of {@link Report reports} and pagination information.
      * @param {Number} pageSize The number of records to return on each page
-     * @param {string} type The type of report to be retrieved
      * @returns {EasyPostObject|Promise<never>} The retrieved {@link EasyPostObject}-based class instance, or a `Promise` that rejects with an error.
      */
-    static async getNextPage(reports, type, pageSize) {
+    static async getNextPage(reports, pageSize) {
+      const type = this._getReportType(reports.reports[0].object);
```

Comment:
Might I recommend instead storing the `type` in the report collection object when it's first retrieved: https://github.com/EasyPost/easypost-csharp/blob/98b378530b8e9234678cfb7ebc7e9ce93f329720/EasyPost/Services/ReportService.cs#L70

And then you have access to it to reuse for subsequent API calls?
https://github.com/EasyPost/easypost-csharp/blob/98b378530b8e9234678cfb7ebc7e9ce93f329720/EasyPost/Services/ReportService.cs#LL82C236-L82C236

This would also need to extend to shipments and trackers, which have additional parameters that need to be stored and reused:
https://github.com/EasyPost/easypost-go/blob/3933664fef2385530e776dc7bda4e1d8a00214eb/shipment.go#L224
https://github.com/EasyPost/easypost-go/blob/3933664fef2385530e776dc7bda4e1d8a00214eb/tracker.go#L227

Pinging @Justintime50 on this too

Reply:
Either seem fine

Reply:
I mean either way works, storing `type` in the report collection will expose `type` which is something undocumented in the API Doc but I don't see any concern with this.

Reply:
Fixed